Source/WebCore/animation/KeyframeEffect.cpp

```diff
@@ -1399,7 +1399,13 @@ void KeyframeEffect::setAnimatedPropertiesInStyle(RenderStyle& targetStyle, doub
     KeyframeValue propertySpecificKeyframeWithZeroOffset(0, RenderStyle::clonePtr(targetStyle));
     KeyframeValue propertySpecificKeyframeWithOneOffset(1, RenderStyle::clonePtr(targetStyle));
 
-    for (auto cssPropertyId : properties) {
+    auto keyframeContainsProperty = [](const KeyframeValue& keyframe, std::variant<CSSPropertyID, AtomString> property) {
+        if (std::holds_alternative<CSSPropertyID>(property))
+            return keyframe.containsProperty(std::get<CSSPropertyID>(property));
+        return keyframe.containsCustomProperty(std::get<AtomString>(property));
```

Comment:
This (and some similar cases below) could use WTF::switchOn instead of holds_alternative/get. That might look nicer.

Reply:
I did not know about this function, but it will definitely improve things! Will fix in commit.